audiomentations/augmentations/equalizer.py

```diff
@@ -0,0 +1,94 @@
+from audiomentations import LowShelfFilter, PeakingFilter, HighShelfFilter
+from audiomentations.core.transforms_interface import BaseWaveformTransform
+
+
+class Equalizer(BaseWaveformTransform):
+    """
+    Adjust the volume of different frequency bands. This transform is a 7-band
```

Comment:
Why is it 7 band? Where is it based from? How is this going to be used?

Equalizers there have different needs for different reasons. For example this paper: 

"Neural Parametric Equalizer Matching Using Differentiable Biquads" by Nercessian they use a 12 band (10 peaking + LS + HS).

In "Neural Third Octave Graphic Equalizer" by Rämö and Välimäki they use 33 bands. 

On the other hand in the IIRNet paper (DIRECT DESIGN OF BIQUAD FILTER CASCADES WITH DEEP LEARNING BY SAMPLING RANDOM POLYNOMIALS by Colonel et al) they use a variable order. 

I would like it to be of parametrized order. 

A suggestion:

Parameters as:

Two booleans for whether we have Lowshelf/Highshelf filters.
An order parameter which controls the number of peaking filters. 
Arrays for center frequencies gains, and qs.

Reply:
> Why is it 7 band? Where is it based from?

It is inspired by Fruity Parametric EQ 2, a plugin that I have personally used a lot, and I like it a lot. It covers all my EQ needs. https://github.com/iver56/audiomentations/issues/12#issuecomment-881243622

> How is this going to be used?

Because it changes the timbre, but keeps the overall "class" of the sound the same, it can be used for data augmentation to make ML models more robust to various frequency spectrums. Many things can affect the spectrum, like room acoustics, any objects that are between the microphone and the sound source, microphone type/model, distance between sound source and microphone...

> Equalizers there have different needs for different reasons. For example this paper:
> 
> "Neural Parametric Equalizer Matching Using Differentiable Biquads" by Nercessian they use a 12 band (10 peaking + LS + HS).
> 
> In "Neural Third Octave Graphic Equalizer" by Rämö and Välimäki they use 33 bands.
> 
> On the other hand in the IIRNet paper (DIRECT DESIGN OF BIQUAD FILTER CASCADES WITH DEEP LEARNING BY SAMPLING RANDOM POLYNOMIALS by Colonel et al) they use a variable order.

Yes, I realize that I cannot/won't cover everyone's EQ needs with this class. It's meant to be a "good enough" general purpose thing.

> parametrized order

Do you mean the number of filters in the chain? Or the order of application?

Reply:
>Yes, I realize that I cannot/won't cover everyone's EQ needs with this class. It's meant to be a "good enough" general purpose thing.

In that case I would be more comfortable with a name that suggests what type of EQ it is.

E.g. `SevenBandGraphicEQ`.

Additionally would be helpful the list of frequencies and Qs in the class docstring since those do not change.

> Do you mean the number of filters in the chain? Or the order of application?

I meant the bands. For example if K=12 the Equalizer can have 

1 LowShelf, + 10 Peaking Filters + 1 Highshelf.

And could center the filters accordingly (so they do not overlap and extend to full band).

A more generic EQ could be better implemented as a transform directly.

Reply:
Ok, I hear you. I think I'll do this:

* Rename it to `SevenBandParametricEQ` (because frequencies and Qs are chosen at random, it's more a parametric EQ than a graphic EQ)
* Add parameters `use_low_shelf_filter` and `use_high_shelf_filter` (bool)
* List min/max center freq of each band in docstring
* Make an issue for a new class `ParametricEQ` with tunable K

Reply:
Sounds like a plan.